modules/python/python-logtemplate.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2018 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "python-logtemplate.h"
+#include "python-logtemplate-options.h"
+#include "python-logmsg.h"
+#include "python-helpers.h"
+#include "scratch-buffers.h"
+#include "messages.h"
+
+PyTypeObject py_log_template_type;
+
+void
+py_log_template_free(PyLogTemplate *self)
+{
+  log_template_unref(self->template);
+  g_free(self->template_options);
+  PyObject_Del(self);
+}
+
+PyObject *
+py_log_template_format(PyObject *s, PyObject *args, PyObject *kwrds)
+{
+  PyLogTemplate *self = (PyLogTemplate *)s;
+
+  PyLogMessage *msg;
+  PyLogTemplateOptions *log_template_options = NULL;
+  gint tz = LTZ_SEND;
+  gint seqnum = 0;
+
+  static const gchar *kwlist[] = {"msg", "template_options", "tz", "seqnum", NULL};
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwrds, "O|Oii", (gchar **)kwlist,
+                                   &msg, &log_template_options, &tz, &seqnum))
+    return NULL;
+
+  if (Py_TYPE(msg) != &py_log_message_type)
+    {
```

Comment:
Please extract the condition into a macro/inline function. The conventions in Python are PySomething_Check(), e.g. PyString_Check().

Reply:
I wasn't trying to suggest a name for the function, only give you a pointer how these functions usually look in CPython. 

I _think_ it would read better if we used our own internal style (e.g. _py_is_log_message or something like that) conventions only use CPython naming when calling CPython or doing some compat functions.

Sorry for the confusion.